taichi/platform/opengl/opengl_data_types.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <taichi/tlang_util.h>
+#include <string>
+
+TLANG_NAMESPACE_BEGIN
+namespace opengl {
+
+inline std::string opengl_data_type_name(DataType dt)
+{
+  // https://www.khronos.org/opengl/wiki/Data_Type_(GLSL)
+  switch (dt) {
+    case DataType::f32:
+      return "float";
+    case DataType::f64:
+      return "double";
+    case DataType::i32:
+      return "int";
+    default:
+      TI_NOT_IMPLEMENTED;
+      break;
+  }
+  return "";
+}
+
+inline std::string opengl_unary_op_type_symbol(UnaryOpType type)
```

Comment:
`unary_op_type_name` is your friend. We can simply use that and just treat some special ops explicitly.